src/gui/guiScrollBar.cpp

```diff
@@ -0,0 +1,454 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2019 stujones11, Stuart Jones <stujones111@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "guiScrollBar.h"
+#include <IGUIButton.h>
+#include <IGUISkin.h>
+
+// Credits: This implementation was based on the original Irrlicht scrollbar.
```

Comment:
I'm pretty sure this means the derivative is still under Irrlicht's license

Reply:
I merely copied this style of crediting from `guiTable`
https://github.com/minetest/minetest/blob/e32a630937de69454c741e93979a7889fcddc7a9/src/gui/guiTable.h#L39-L42
However, I am happy to change the license headers if someone tells me how it should be done.

Reply:
Looking again, I see `guiTable` only really copies the interface, I have now corrected it, just to be sure.